src/main/java/tira/Storage.java

```diff
@@ -0,0 +1,152 @@
+package tira;
+
+import tira.task.Task;
+import tira.task.Deadline;
+import tira.task.ToDo;
+import tira.task.Event;
+
+import java.util.ArrayList;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.FileReader;
+import java.io.FileNotFoundException;
+import java.io.File;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.io.BufferedReader;
+
+public class Storage {
+    private static DateTimeFormatter OUT_FORMATTER = DateTimeFormatter.ofPattern("dd MMM yyyy");
+    private final String filePath;
+    private final Ui ui = new Ui();
+
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    /*
+    Solution below inspired by https://github.com/hansneddyanto/ip/blob/master/src/main/java/hana/Storage.java
+     */
+    public ArrayList<Task> load() throws TiraException {
+        ArrayList<Task> tasks = new ArrayList<Task>();
+        File file = new File(filePath); //try creating filePath
+        FileReader fileReader = null;
+
+        try {
+            if (!file.exists()) { //create new file if the file doesn't exist
+                file.getParentFile().mkdirs();
+                file.createNewFile();
+                return tasks;
+            } else {// a file exists
+                fileReader = new FileReader(file);
+                BufferedReader bufferedReader = new BufferedReader(fileReader);
+                ArrayList<String> lines = new ArrayList<>();
+                String line = bufferedReader.readLine();
+                while (line != null) {
+                    lines.add(line);
+                    if (bufferedReader.readLine() == null) {
+                        break;
+                    }
+                }
+                return this.convertToTaskList(lines);
+            }
```

Comment:
I like how you handled the case where the file doesn't exist at the start of this try block! :+1:
Perhaps it might be beneficial to extract the code from within the `else` block, since it is the expected normal flow of the program, I feel that this could improve the readability of the code.
```suggestion
            if (!file.exists()) { //create new file if the file doesn't exist
                file.getParentFile().mkdirs();
                file.createNewFile();
                return tasks;
            }
            
            // a file exists
            fileReader = new FileReader(file);
            BufferedReader bufferedReader = new BufferedReader(fileReader);
            ArrayList<String> lines = new ArrayList<>();
            String line = bufferedReader.readLine();
            while (line != null) {
                lines.add(line);
                if (bufferedReader.readLine() == null) {
                    break;
                }
            }
            return this.convertToTaskList(lines);
```